src/spacel/provision/template/app.py

```diff
@@ -74,20 +73,32 @@ def app(self, app, region):
             app_hostname = '%s-%s.%s' % (app.name, orbit.name, orbit.domain)
             params['VirtualHost']['Default'] = app_hostname
 
-        # Expand ELB to all AZs:
-        public_elb_subnets = orbit.public_elb_subnets(region)
-        private_elb_subnets = orbit.private_elb_subnets(region)
-        self._subnet_params(params, 'PublicElb', public_elb_subnets)
-        self._subnet_params(params, 'PrivateElb', private_elb_subnets)
+        if app.loadbalancer == 'true':
+            params['ElbScheme']['Default'] = app.scheme
+            # Expand ELB to all AZs:
+            public_elb_subnets = orbit.public_elb_subnets(region)
+            private_elb_subnets = orbit.private_elb_subnets(region)
+            self._subnet_params(params, 'PublicElb', public_elb_subnets)
+            self._subnet_params(params, 'PrivateElb', private_elb_subnets)
 
-        self._elb_subnets(resources, 'PublicElb', public_elb_subnets)
-        self._elb_subnets(resources, 'PrivateElb', private_elb_subnets)
+            self._elb_subnets(resources, 'PublicElb', public_elb_subnets)
+            self._elb_subnets(resources, 'PrivateElb', private_elb_subnets)
 
         # Expand ASG to all AZs:
-        private_instance_subnets = orbit.private_instance_subnets(region)
-        self._subnet_params(params, 'PrivateInstance', private_instance_subnets)
-        self._asg_subnets(resources, 'PrivateInstance',
-                          private_instance_subnets)
+        if app.availability == 'public':
+            public_instance_subnets = orbit.public_instance_subnets(region)
+            self._subnet_params(params, 'PrivateInstance',
```

Comment:
I think I understand the point you're making about using private for public, but the proposed solution doesn't make sense yet.

I was thinking to just go the full Public vs Private route and create and destroy them as needed, separately. But maybe I'm just missing something obvious to make this easier/better?